### PR TITLE
refactor: backend structure improvements

### LINKED ISSUE #60: Backend Structure Improvements
This exists to look at the first implemented backend structure by domi in #58.
It is more research than actual implementation, maybe with some improvements.

### PR DESCRIPTION
**Pull request by theoleuthardt**
# Summary
This Pull Request makes improvements to the codebase by standardizing key attribute changes, reorganizing file structures, and cleaning up redundant code.

## Standardization:
Replaced id with backlogEntryID and similar changes to align with consistent naming conventions in the frontend components.

## Refactoring:
Enhanced readability and reusability by transitioning from positional parameters to object parameters in service functions (e.g., updateBacklogEntry, createBacklogEntry).
Simplified repetitive CRUD operations by wrapping them in modular abstractions.

## Reorganization:
Consolidated types and constant imports under a new structure at ~/app/types to declutter dependencies.

## Error Handling:
Introduced custom error types and unified error handling across database operations.

## Code Style:
Improved function readability with explicit array array destructuring, better indentation, and variable renaming.

## Test Updates:
Adjusted test cases to align with the updated service and data model structures.

---
Enhanced coverage for error handling and edge cases.

### CODE REVIEW

File: src/server/db/types/errors.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      message,
+    });
+  }
+}
+
+/**
+ * Map PostgreSQL error codes to appropriate tRPC errors
+ */
+export function handleDatabaseError(error: any, operation: string): never {
+  // PostgreSQL error codes
```
Comment (AuriomTex) [inline src/server/db/types/errors.ts]: Maybe remove inline comments

File: src/server/db/utils/transaction.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * Automatically handles BEGIN, COMMIT, and ROLLBACK
+ *
+ * @example
+ * const result = await withTransaction(pool, async (client) => {
+ *   await client.query('INSERT INTO ...');
+ *   await client.query('UPDATE ...');
+ *   return { success: true };
+ * });
+ */
+export async function withTransaction<T>(
```
Comment (AuriomTex) [inline src/server/db/utils/transaction.ts:15]: transaction used for error handling?

File: test/howLongToBeat.test.ts [hunk 3]
```diff
@@ -1,18 +1,78 @@
...
+describe("HowLongToBeat Integration", () => {
+  describe("SearchGame", () => {
+    it("should find game by title", async () => {
+      const result = await SearchGame("Fortnite");
+      expect(result).toBeDefined();
+      expect(result.length).toBeGreaterThan(0);
+      expect(result[0]?.title).toBe("Fortnite");
+    });
 
-        it('Gooby', async () => {
```
Comment (AuriomTex) [inline test/howLongToBeat.test.ts:6]: Bye Bye Gooby Fortnite

**Review by AuriomTex — APPROVED** [review]
Since everything works, the changes are looking nice.
I dont understand some decisions, but we can discuss at a meeting :)

File: src/server/db/utils/transaction.ts (hunk 2 above)
Comment (theoleuthardt) [reply src/server/db/utils/transaction.ts:15]: # Explanation:
You need transaction.ts for Postgres to ensure
data consistency and integrity across multiple
database operations. Here's why:

## What Database Transactions Do

A transaction groups multiple database
operations into a single atomic unit - either
all operations succeed or none do. This
prevents partial updates that could leave your
data in an inconsistent state.

## Why Your File Exists

Your transaction.ts (at
src/server/db/utils/transaction.ts) provides
two helper functions:

1. withTransaction - Wraps operations in
BEGIN/COMMIT/ROLLBACK automatically
2. executeInTransaction - Simplified API for
running multiple queries together

## Real-World Example

### Without transactions, this scenario could corrupt your data:
``` typescript
// Delete a backlog entry and its related data
await db.query('DELETE FROM backlog_entries 
WHERE id = $1', [id]);
// ❌ If this fails, the entry is deleted but categories remain orphaned
await db.query('DELETE FROM entry_categories WHERE entry_id = $1', [id]);
```

### With your transaction helper:

```typescript
await executeInTransaction(pool, 'deleteEntry',
 [
  { query: 'DELETE FROM backlog_entries WHERE 
id = $1', params: [id] },
  { query: 'DELETE FROM entry_categories WHERE 
entry_id = $1', params: [id] }
]);
// ✅ Both succeed or both fail - no orphaned data
```

### Key Benefits

- Atomicity: All-or-nothing execution
- Consistency: Database stays in valid state
- Isolation: Concurrent operations don't
interfere
- Error handling: Auto-rollback on failure
- Resource management: Client connection is
always released

---
This pattern is essential for any multi-step
database operation where consistency matters.

File: src/server/db/types/errors.ts (hunk 1 above)
Comment (theoleuthardt) [reply src/server/db/types/errors.ts]: removed
